arch/arm/core/aarch32/irq_manage.c

```diff
@@ -74,7 +74,13 @@ void z_arm_irq_priority_set(unsigned int irq, unsigned int prio, uint32_t flags)
 	 * via flags
 	 */
 	if (IS_ENABLED(CONFIG_ZERO_LATENCY_IRQS) && (flags & IRQ_ZERO_LATENCY)) {
+#if defined(CONFIG_ZERO_LATENCY_LEVELS)
+		if (CONFIG_ZERO_LATENCY_LEVELS == 1) {
+			prio = _EXC_ZERO_LATENCY_IRQS_PRIO;
+		}
+#else
 		prio = _EXC_ZERO_LATENCY_IRQS_PRIO;
+#endif
```

Comment:
Suggestion,

```suggestion
		if (ZERO_LATENCY_LEVELS == 1) {
			prio = _EXC_ZERO_LATENCY_IRQS_PRIO;
		} else {
			/* Use caller supplied prio level as-is */
		}
```

and have at the top of the file

```c
#if defined(CONFIG_ZERO_LATENCY_LEVELS)
#define ZERO_LATENCY_LEVELS CONFIG_ZERO_LATENCY_LEVELS
#else
#define ZERO_LATENCY_LEVELS 1
#endif 
```